graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/FederationDirectives.java

```diff
@@ -171,98 +143,36 @@ public static GraphQLDirective provides(String fields) {
-          .inputValueDefinition(
-              newInputValueDefinition()
-                  .name("url")
-                  .type(new NonNullType(new TypeName("String")))
-                  .build())
-          .inputValueDefinition(
-              newInputValueDefinition()
-                  .name("import")
-                  .type(new ListType(new TypeName("link__Import")))
-                  .build())
-          .repeatable(true)
-          .build();
-
-  /* directive @tag(name: String!) repeatable on FIELD_DEFINITION | INTERFACE | OBJECT | UNION */
-  public static final String tagName = "tag";
-
-  public static final DirectiveDefinition tagDefinition =
-      newDirectiveDefinition()
-          .name(tagName)
-          .inputValueDefinition(
-              newInputValueDefinition()
-                  .name("name")
-                  .type(new NonNullType(new TypeName("String")))
-                  .build())
-          .directiveLocations(Arrays.asList(DL_FIELD_DEFINITION, DL_INTERFACE, DL_OBJECT, DL_UNION))
-          .repeatable(true)
-          .build();
-
-  // directive @override(from: String!) on FIELD_DEFINITION
-  public static final DirectiveDefinition overrideDefinition =
-      newDirectiveDefinition()
-          .name("override")
-          .inputValueDefinition(
-              newInputValueDefinition()
-                  .name("from")
-                  .type(new NonNullType(new TypeName("String")))
-                  .build())
-          .directiveLocations(Arrays.asList(DL_FIELD_DEFINITION))
-          .build();
-
-  // directive @inaccessible on FIELD_DEFINITION | INTERFACE | OBJECT | UNION | ARGUMENT_DEFINITION
-  // | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION
-  public static final DirectiveDefinition inaccessibleDefinition =
-      newDirectiveDefinition()
-          .name("inaccessible")
-          .directiveLocations(
-              Arrays.asList(
-                  DL_FIELD_DEFINITION,
-                  DL_INTERFACE,
-                  DL_OBJECT,
-                  DL_UNION,
-                  DL_ARGUMENT_DEFINITION,
-                  DL_SCALAR,
-                  DL_ENUM,
-                  DL_ENUM_VALUE,
-                  DL_INPUT_OBJECT,
-                  DL_INPUT_FIELD_DEFINITION))
-          .build();
-
   private FederationDirectives() {}
 
   /* Sets */
 
   public static final Set<String> allNames;
   public static final Set<GraphQLDirective> allDirectives;
   public static final Set<DirectiveDefinition> allDefinitions;
-  public static final Set<DirectiveDefinition> federation2DirectiveDefinitions;
+  static final List<SDLNamedDefinition> federation2Definitions;
   public static final Set<DirectiveDefinition> federation1DirectiveDefinitions;
 
+  private static List<SDLNamedDefinition> fed2Definitions() {
+    InputStream inputStream =
+        FederationDirectives.class.getClassLoader().getResourceAsStream("fed2directives.graphqls");
```

Comment:
Very cool, a lot simpler / less verbose!

Reply:
Pretty happy about that part :)